app/frontend/src/search/ui/input/location.js

```diff
@@ -0,0 +1,26 @@
+import { setGeocodeAttributes, removeGeocodeAttributes } from '../form';
+import { enableRadiusSelect, disableRadiusSelect } from '../radius';
+import { stringMatchesPostcode } from '../../../lib/utils';
+import { getGeolocatedCoordinates } from '../../../lib/api';
+
+export const onSubmit = (query, locations, client) => {
+    if (shouldGeocode(query, locations)) {
+        getGeolocatedCoordinates(query).then(coords => {
+            geocodeSuccess(coords, client);
+        });
+    } else {
+        disableRadiusSelect();
+        removeGeocodeAttributes();
+        client.refresh();
+    }
+};
+
+export const shouldGeocode = (query, locations) => stringMatchesPostcode(query) || (query.length && locations.indexOf(query.toLowerCase()) === -1);
```

Comment:
To me this logic would be simpler if the function was actually `shouldNotGeocode`. Then we only need to check the presence of the location in the list.